src/main/java/com/microsoft/graph/requests/ResponseBodyHandler.java

```diff
@@ -0,0 +1,95 @@
+import com.microsoft.graph.exceptions.ServiceException;
+import com.microsoft.kiota.ApiException;
+import com.microsoft.kiota.serialization.*;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.*;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A response handler for deserializing responses to a ModelType. Particularly for Batch requests.
+ * @param <T> the ModelType of the response body.
+ */
+public class ResponseBodyHandler<T extends Parsable> implements com.microsoft.kiota.ResponseHandler {
+    private final ParseNodeFactory parseNodeFactory;
+    private final ParsableFactory<T> factory;
+    /**
+     * Instantiates a new response handler.
+     * @param parseNodeFactory the parse node factory to use when deserializing the response.
+     * @param factory the factory to use when deserializing the response to a ModelType.
+     */
+    public ResponseBodyHandler(@Nullable ParseNodeFactory parseNodeFactory, @Nonnull ParsableFactory<T> factory) {
+        this.parseNodeFactory = (parseNodeFactory == null) ? ParseNodeFactoryRegistry.defaultInstance : parseNodeFactory;
+        this.factory = factory;
+    }
+    /**
+     * Instantiates a new response handler.
+     * @param factory the factory to use when deserializing the response to a ModelType.
+     */
+    public ResponseBodyHandler(@Nonnull ParsableFactory<T> factory) {
+        this(null, factory);
+    }
+    /**
+     * Handles the response and returns the deserialized response body as a ModelType.
+     * @param response The native response object.
+     * @param errorMappings the error mappings for the response to use when deserializing failed responses bodies. Where an error code like 401 applies specifically to that status code, a class code like 4XX applies to all status codes within the range if an the specific error code is not present.
+     * @return A CompletableFuture that represents the asynchronous operation and contains the deserialized response.
+     * @param <NativeResponseType> The type of the native response object.
+     * @param <ModelType> The type of the response model object.
+     */
+    @Nonnull
+    @Override
+    @SuppressFBWarnings
+    public <NativeResponseType, ModelType> CompletableFuture<ModelType> handleResponseAsync(@Nonnull NativeResponseType response, @Nullable HashMap<String, ParsableFactory<? extends Parsable>> errorMappings) {
+        if(response instanceof Response && ((Response) response).body()!=null) {
+            Response nativeResponse = (Response) response;
+            ResponseBody body = nativeResponse.body();
+            try(final InputStream in = body.byteStream()) {
+                byte[] responseStreamBytes = ByteStreams.toByteArray(in); //Used to hold a copy of the response stream for error handling.
```

Comment:
I don't think you need a copy here since you're reusing the same parse not object. You can pass the input stream directly the the parse not factory.

Reply:
Ah yes, good catch. I restructured the methods to use the same ParseNode when I was removing the repetitive code and forgot to remove this line. Thanks!